care/facility/api/viewsets/asset.py

```diff
@@ -98,6 +98,12 @@ class AssetFilter(filters.FilterSet):
     qr_code_id = filters.CharFilter(field_name="qr_code_id", lookup_expr="icontains")
 
 
+class AssetPublicViewSet(RetrieveModelMixin, GenericViewSet):
+    queryset = Asset.objects.all()
+    serializer_class = AssetSerializer
+    lookup_field = "external_id"
+    permission_classes = [IsAuthenticatedOrReadOnly]
```

Comment:
We can leave this empty since this is a read only endpoint.

Reply:
Done